From a2fd63964c69e53108725396c7736b80c956a4f4 Mon Sep 17 00:00:00 2001
From: Chien Wong <m@xv97.com>
Date: Fri, 5 Jan 2024 17:50:58 +0800
Subject: [PATCH] build: bgscan_simple depends on WNM

Fix build of wpa_supplicant if bgscan_simple is enabled while
WNM is not, as in the defconfig.

Fixes: ceb7f65dcc94 ("bgscan: Allow simple bgscan to do BTM queries")
Signed-off-by: Chien Wong <m@xv97.com>
---
 wpa_supplicant/Android.mk | 4 ++++
 wpa_supplicant/Makefile   | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/wpa_supplicant/Android.mk b/wpa_supplicant/Android.mk
index 809973a21..287be70cb 100644
--- a/wpa_supplicant/Android.mk
+++ b/wpa_supplicant/Android.mk
@@ -305,6 +305,10 @@ ifdef CONFIG_MBO
 CONFIG_WNM=y
 endif
 
+ifdef CONFIG_BGSCAN_SIMPLE
+CONFIG_WNM=y
+endif
+
 ifdef CONFIG_WNM
 L_CFLAGS += -DCONFIG_WNM
 OBJS += wnm_sta.c
diff --git a/wpa_supplicant/Makefile b/wpa_supplicant/Makefile
index eccfd68c5..ff4237032 100644
--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -338,6 +338,10 @@ ifdef CONFIG_MBO
 CONFIG_WNM=y
 endif
 
+ifdef CONFIG_BGSCAN_SIMPLE
+CONFIG_WNM=y
+endif
+
 ifdef CONFIG_WNM
 CFLAGS += -DCONFIG_WNM
 OBJS += wnm_sta.o
-- 
2.43.0

